### PR TITLE
Clean-up Phase #2

### PR DESCRIPTION
**Pull request by isfaaghyth**
- [x] add test-case for feature of dashboard
- [x] put runtime permission checker on baseActivity
- [x] clean-up: line of code on UI classes
- [x] defined an any static variable on `companion object`
- [x] created extension for getter of color
- [x] removed unused resources

### CODE REVIEW

File: app/src/main/java/id/rizmaulana/covid19/data/model/CovidDaily.kt [hunk 1]
```diff
@@ -12,10 +12,10 @@ data class CovidDaily(
     @Expose @SerializedName("deltaConfirmed") val deltaConfirmed: Int = 0,
     @Expose @SerializedName("deltaRecovered") val deltaRecovered: Int = 0,
     @Expose @SerializedName("mainlandChina") val mainlandChina: Int = 0,
-    @Expose @SerializedName("objectid") val objectid: Int = 0,
+    @Expose @SerializedName("objectId") val objectId: Int = 0,
```
Comment (rizmaulana) [inline app/src/main/java/id/rizmaulana/covid19/data/model/CovidDaily.kt]: Ini kayaknya SerializedName-nya tetap gini mas @isfaaghyth `@SerializedName("objectid")`

File: app/src/main/java/id/rizmaulana/covid19/data/model/CovidDaily.kt [hunk 2]
```diff
@@ -12,10 +12,10 @@ data class CovidDaily(
...
     @Expose @SerializedName("otherLocations") val otherLocations: Int = 0,
     @Expose @SerializedName("reportDate") val reportDate: Long = 0,
-    @Expose @SerializedName("reportDateString") val reportDateString: String? = null,
+    @Expose @SerializedName("reportDateString") val reportDateString: String = "",
```
Comment (rizmaulana) [inline app/src/main/java/id/rizmaulana/covid19/data/model/CovidDaily.kt]: Afaik, ini kalau data dari API-nya `null` string tetap jadi `null` mas @isfaaghyth 
Reference : https://proandroiddev.com/safe-parsing-kotlin-data-classes-with-gson-4d560fe3cdd2

Kalau benar2 perlu gak nullable, supaya safe biasanya pakai custom getter 

```suggestion
    @Expose @SerializedName("reportDateString") private val _reportDateString: String? = null,
    ...
    val reportDateString get() = _reportDateString.orEmpty()
```

File: app/src/main/java/id/rizmaulana/covid19/data/model/CovidDetail.kt [hunk 3]
```diff
@@ -9,15 +9,14 @@ import kotlinx.android.parcel.Parcelize
  * rizmaulana 04/03/20.
  */
 
-@Parcelize
-data class CovidDetail(
+@Parcelize data class CovidDetail(
     @Expose @SerializedName("confirmed") val confirmed: Int = 0,
-    @Expose @SerializedName("countryRegion") val countryRegion: String? = null,
+    @Expose @SerializedName("countryRegion") val countryRegion: String = "",
```
Comment (rizmaulana) [inline app/src/main/java/id/rizmaulana/covid19/data/model/CovidDetail.kt]: Refer to https://github.com/rizmaulana/kotlin-mvvm-covid19/pull/2#discussion_r390191156

File: app/src/main/java/id/rizmaulana/covid19/data/model/CovidOverview.kt [hunk 4]
```diff
@@ -7,7 +7,7 @@ import com.google.gson.annotations.SerializedName
  * rizmaulana 04/03/20.
  */
 data class CovidOverview(
-    @Expose @SerializedName("confirmed") val confirmed: CovidOverviewItem? = null,
-    @Expose @SerializedName("recovered") val recovered: CovidOverviewItem? = null,
-    @Expose @SerializedName("deaths") val deaths: CovidOverviewItem? = null
+    @Expose @SerializedName("confirmed") val confirmed: CovidOverviewItem = CovidOverviewItem(),
```
Comment (rizmaulana) [inline app/src/main/java/id/rizmaulana/covid19/data/model/CovidOverview.kt]: Refer to https://github.com/rizmaulana/kotlin-mvvm-covid19/pull/2#discussion_r390191156

File: app/src/main/java/id/rizmaulana/covid19/data/model/CovidOverviewItem.kt [hunk 5]
```diff
@@ -8,6 +8,6 @@ import com.google.gson.annotations.SerializedName
  * rizmaulana 04/03/20.
  */
 data class CovidOverviewItem(
-    @Expose @SerializedName("detail") val detail: String? = null,
+    @Expose @SerializedName("detail") val detail: String = "",
```
Comment (rizmaulana) [inline app/src/main/java/id/rizmaulana/covid19/data/model/CovidOverviewItem.kt]: Refer to https://github.com/rizmaulana/kotlin-mvvm-covid19/pull/2#discussion_r390191156

File: app/src/main/java/id/rizmaulana/covid19/data/model/CovidDaily.kt (hunk 2 above)
Comment (isfaaghyth) [reply app/src/main/java/id/rizmaulana/covid19/data/model/CovidDaily.kt]: ok noted!

File: app/src/main/java/id/rizmaulana/covid19/data/model/CovidDaily.kt (hunk 1 above)
Comment (isfaaghyth) [reply app/src/main/java/id/rizmaulana/covid19/data/model/CovidDaily.kt]: wah iya, keikut name refactor, fixing..

File: app/src/main/java/id/rizmaulana/covid19/ui/overview/DashboardViewModel.kt [hunk 6]
```diff
@@ -38,6 +37,7 @@ class DashboardViewModel(
 
     fun getOverview() {
         appRepository.overview()
+            .subscribeOn(schedulerProvider.io())
```
Comment (rizmaulana) [inline app/src/main/java/id/rizmaulana/covid19/ui/overview/DashboardViewModel.kt]: Ini sebetulnya sudah ada scheduler ke `createWithScheduler(Schedulers.io())` waktu instantiate mas @isfaaghyth 
ada di file `NetworkModule.kt` 


```
single {
        Retrofit.Builder()
            .baseUrl(BuildConfig.BASE_URL)
            .client(get<OkHttpClient>())
            .addCallAdapterFactory(RxJava2CallAdapterFactory.createWithScheduler(Schedulers.io()))
            .addConverterFactory(GsonConverterFactory.create())
            .build()

    }
```

Comment (isfaaghyth) [reply app/src/main/java/id/rizmaulana/covid19/ui/overview/DashboardViewModel.kt]: ah I see, removed, mas! sorry sorry haha..

Comment (rizmaulana) [reply app/src/main/java/id/rizmaulana/covid19/ui/overview/DashboardViewModel.kt]: Siap mas hehe

**Pull request merged**